verticapy/learn/causal_model.py

```diff
@@ -0,0 +1,343 @@
+# import
+from verticapy import vDataFrame, vDataFrameSQL
+from verticapy.toolbox import check_types
+import verticapy as vp
+import itertools
+from tqdm.auto import tqdm
+from time import time
+from typing import Union
+
+import warnings
+
+from ..stats.tools import conditional_chi_square, is_dir_path
+import numpy as np
+from verticapy import plot
+
+
+# ---#
+class CausalModel:
+	"""
+-------------------------------------------------
+Main Class for Causal Models.
+
+Parameters
+----------
+input_relation: [str, vDataframe]
+	Name of original relation
+	"""	
+ 
+	# ---#
+	def __init__(self, input_relation : Union[str, vDataFrame]):
+		check_types([("input_relation", input_relation, [str, vDataFrame])])
+		vdf = input_relation
+		if isinstance(vdf, str):
+			vdf = vDataFrameSQL(relation=vdf)
+
+		self.input_relation = vdf
+		self.vdf = vdf.copy()
```

Comment:
Why do we need two copies? Perhaps, you can add a short comment to briefly explain what are `self.input_relation` and `self.vdf` and why both should be available.

Reply:
I wanted to have two copies in case a user wanted to change the discretization parameters, that way we didn't lose the original relationship. So `self.input_relation` is the main file and the function operates on `self.vdf` which can be modified by the discretization function and the method itself.

Reply:
@zacandcheese can you please add a comment in your code